saml-authentication-server/src/main/java/jetbrains/buildServer/auth/saml/plugin/SamlAuthenticationScheme.java

```diff
@@ -139,6 +144,22 @@ public HttpAuthenticationResult processAuthenticationRequest(@NotNull HttpServle
 
                             if (user == null) {
                                 LOG.warn(String.format("New user %s was not created due to unknown reason", username));
+                            } else {
+                                String groups = getAttribute(auth, settings.getGroupsAttributeMapping());
+                                LOG.debug(String.format("SAML Groups = '%s'", groups));
+
+                                // Look for matching userGroup
+                                String[] assignedGroups = groups.split(",");
+                                for (String group: assignedGroups) {
+                                    SUserGroup matchingGroup = userGroups.findUserGroupByName(group);
```

Comment:
Meanwhile this works, I would rather do it bit differently - I would fetch the list of all user groups names into a Hashset<STring> before the loop and just use it as a lookup. It has another advantage - we could additionaly lowercase/uppercase groups in the SAML attribute and group names from Teamcity and make the mapping case-insensitive.

Reply:
Addressed in [`732591b` (#52)](https://github.com/morincer/teamcity-plugin-saml/pull/52/commits/732591b0503ddf1447ad56eb81d5497aad25b4b0)